File: algorithms/cpp/474.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int findMaxForm(vector<string> &strs, int m, int n) {
        int dp[101][101] = {0};
        for (const auto &s: strs) {
            int zeros = count(s.begin(), s.end(), '0');
            int ones = s.size() - zeros;
            for (int i = m; i >= zeros; i--)
                for (int j = n; j >= ones; j--)
                    dp[i][j] = max(dp[i][j], dp[i-zeros][j-ones] + 1);
        }
        return dp[m][n];
    }
};

int main() {
    return 0;
}

File: algorithms/python/474.py
class Solution:
    def findMaxForm(self, strs, m, n):
        """
        :type strs: List[str]
        :type m: int
        :type n: int
        :rtype: int
        """
        strs.sort(key=len)
        dic = {}
        dic[(0, 0)] = 0
        for s in strs:
            zeroes = s.count('0')
            ones = len(s) - zeroes
            tmp = {}
            for k, v in dic.items():
                tmp[k] = v
                z, o = k
                if z + zeroes > m or o + ones > n:
                    continue
                tmp[(z+zeroes, o+ones)] = max(dic.get((z+zeroes, o+ones), 0), v + 1)
            dic = tmp
        return max(dic.values())
